provider/resource_processor.go

```diff
@@ -130,3 +131,23 @@ func resourceProcessorDelete(d *schema.ResourceData, meta any) error {
 func resourceProcessorImportState(_ context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
 	return genericResourceImport(model.KindProcessor, d, meta)
 }
+
+// validateParametersJSON validates the parameters_json field during plan phase
+func validateParametersJSON(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
nit. Im sure interface{} is used all over the place in here.

```suggestion
func validateParametersJSON(v any, k string) (ws []string, errors []error) {
```